internal/sys/types.go

```diff
@@ -490,18 +491,32 @@ type BtfInfo struct {
 	KernelBtf uint32
 }
 
+func (s *BtfInfo) Pin(p *runtime.Pinner) {
+	p.Pin(s)
+	p.Pin(&s.Btf)
```

Comment:
Ah, this is subtle. We don't want to pin `&s.Btf` which is the address of the `Btf` member of this struct. Instead we need to pin the pointer it contains, which is in `s.Btf.ptr`.

How about implementing a pin function on `Pointer`? `func (p Pointer) Pin(*runtime.Pinner)`.